### ISSUE TITLE
DocFX incorrectly outputs href issues in referenced TOC files

### ISSUE THREAD

**Issue by winseros** [dotnet/docfx#3026]
**DocFX version**: 2.37

**Template used**: no templates

**Steps to Reproduce**:

##### The following folder structure
```
+-- a1
|   +-- toc.yml
+-- a2
|   +-- toc.yml
+-- docfx.json
```

##### With files
*a1/toc.yml*
```
- name: TOC2
  href: ../a2/toc.yml
```

*a2/toc.yml*
```
- name: Article2
  href: not-existing2.md
```

*docfx.json*
```
{
  "build": {
    "content": [
      {
        "files": [
          "**/*.md",
          "**/**.yml"
        ],
        "exclude": [
          "**/obj/**",
          "**/includes/**",
          "README.md",
          "LICENSE",
          "LICENSE-CODE",
          "ThirdPartyNotices",
          "PULL_REQUEST_TEMPLATE.md",
          "ISSUE_TEMPLATE.md",
          "**/_themes/**",
          "**/_themes.pdf/**"
        ]
      }
    ],
    "resource": [
      {
        "files": [
          "**/**.png",
          "**/**.jpg",
          "**/**.jpeg",
          "**/**.gif",
          "**/**.svg"
        ],
        "exclude": [
          "**/obj/**",
          "**/includes/**"
        ]
      }
    ],
    "overwrite": [],
    "externalReference": [],
    "globalMetadata": {
    },
    "fileMetadata": {
    },
    "template": [],
    "dest": "d365F-O",
    "markdownEngineName": "markdig"
  }
}
```

##### Run
`docfx build <path_to_docfx.json>`

**Expected Behavior**:
Expected docfx to output the following errors regarding the missing `not-existing2.md` at `a2/toc.yml`
```
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.TocDocumentProcessor.Save](a2/toc.yml)Invalid file link:(not-existing2.md)
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateHref](a2/toc.yml)Unable to find file "not-existing2.md" for Href referenced by TOC file "a2/toc.yml"
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateHref](a2/toc.yml)Unable to find file "not-existing2.md" for TopicHref referenced by TOC file "a2/toc.yml"
```

**Actual Behavior**:
docfx outputs:
```
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.TocDocumentProcessor.Save](a1/toc.yml)Invalid file link:(~/a2/not-existing2.md)
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateHref](a1/toc.yml)Unable to find file "../a2/not-existing2.md" for Href referenced by TOC file "a1/toc.yml"
[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateHref](a1/toc.yml)Unable to find file "../a2/not-existing2.md" for TopicHref referenced by TOC file "a1/toc.yml"
```

**In other words**
If your docs contain toc file A referencing another toc file B, then any incorrect hrefs in B are reported against A, but not against B as they should. On huge docs bases that confuses a lot and makes it very hard to find a particular href issue.

Thx

**Comment by herohua** [comment]
@OsmondJiang @live1206

**Comment by superyyrrzz** [comment]
@winseros Thank you for the report and detailed repro steps. I agree the warning message is not accurate under toc inclusion case. More information should be added to the toc model to fix this.

@OsmondJiang Has v3 the ideal behavior?

**Comment by OsmondJiang** [comment]
@superyyrrzz should have the same problem, i am working on a fix. What about the inclusion error behavior @yufeih, @superyyrrzz ?

**Comment by winseros** [comment]
@OsmondJiang, Thank you!
Is the fix expected to be for v3 or for v2 also?

**Comment by OsmondJiang** [comment]
@winseros , the fix is only for v3 now, @superyyrrzz  is working on a fix for v2 now. :)

**Comment by yufeih** [comment]
error is reported correctly for markdown inclusion, there is a test case for that in v3

**Comment by superyyrrzz** [comment]
@winseros I am investigating on this. It should need some work on v2.

--- tech details ---
Warning is logged in `save` phase: https://github.com/dotnet/docfx/blob/2eddbfd0e6ebb250782e926d5b41d1d6cba0a789/src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandler.cs#L177

This fix need to provide information in `FileLinkSources` for a correct warning.
1. Save included TOC's path in TocDocumentProcessor's prebuild
2. Save FileLinkSource in TocDocumentProcessor's build

**Comment by herohua** [comment]
@winseros , fix released in [v2.38](https://github.com/dotnet/docfx/releases/tag/v2.38)
